### ISSUE TITLE
Cannot install the dependencies as given in the GPT-2 tutorial - please fix the versions

### ISSUE THREAD

**Issue by AayushSameerShah** [microsoft/onnxruntime#16515]
### Describe the documentation issue

I am having **2** issues:

# 1️⃣ The dependencies

I was following this tutorial: ***onnxruntime/python/tools/transformers/notebooks/Inference_GPT2_with_OnnxRuntime_on_CPU.ipynb*** ([link](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/notebooks/Inference_GPT2_with_OnnxRuntime_on_CPU.ipynb)) but the code to install the `torch` and `torchvision` versions are not proper:

```
!{sys.executable} -m pip install torch==1.10.1+cpu torchvision==0.11.2+cpu torchaudio==0.10.1 -f https://download.pytorch.org/whl/torch_stable.html --no-warn-script-location >pip_output.txt
```

Returns:
```
ERROR: Could not find a version that satisfies the requirement torch==1.10.1+cpu (from versions: 1.11.0, 1.11.0+cpu, 1.11.0+cu102, 1.11.0+cu113, 1.11.0+cu115, 1.11.0+rocm4.3.1, 1.11.0+rocm4.5.2, 1.12.0, 1.12.0+cpu, 1.12.0+cu102, 1.12.0+cu113, 1.12.0+cu116, 1.12.0+rocm5.0, 1.12.0+rocm5.1.1, 1.12.1, 1.12.1+cpu, 1.12.1+cu102, 1.12.1+cu113, 1.12.1+cu116, 1.12.1+rocm5.0, 1.12.1+rocm5.1.1, 1.13.0, 1.13.0+cpu, 1.13.0+cu116, 1.13.0+cu117, 1.13.0+cu117.with.pypi.cudnn, 1.13.0+rocm5.1.1, 1.13.0+rocm5.2, 1.13.1, 1.13.1+cpu, 1.13.1+cu116, 1.13.1+cu117, 1.13.1+cu117.with.pypi.cudnn, 1.13.1+rocm5.1.1, 1.13.1+rocm5.2, 2.0.0, 2.0.0+cpu, 2.0.0+cpu.cxx11.abi, 2.0.0+cu117, 2.0.0+cu117.with.pypi.cudnn, 2.0.0+cu118, 2.0.0+rocm5.3, 2.0.0+rocm5.4.2, 2.0.1, 2.0.1+cpu, 2.0.1+cpu.cxx11.abi, 2.0.1+cu117, 2.0.1+cu117.with.pypi.cudnn, 2.0.1+cu118, 2.0.1+rocm5.3, 2.0.1+rocm5.4.2)
ERROR: No matching distribution found for torch==1.10.1+cpu
```

# 2️⃣ Can not export the model
There is a cell which has this code:
```sh
!{sys.executable} -m onnxruntime.transformers.models.gpt2.convert_to_onnx -m $model_name_or_path --output $onnx_model_path -o -p fp32 --use_int32_inputs -t 10>export_output.txt 2>&1
```

Results:
```
2023-06-28 07:39:03.648074: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
usage: convert_to_onnx.py
       [-h]
       -m
       MODEL_NAME_OR_PATH
       [--model_class {GPT2LMHeadModel,GPT2LMHeadModel_NoPadding,GPT2Model}]
       [--cache_dir CACHE_DIR]
       [--output OUTPUT]
       [-o]
       [--use_gpu]
       [--provider {dml,rocm,migraphx,cuda,tensorrt}]
       [--tolerance TOLERANCE]
       [--input_test_file INPUT_TEST_FILE]
       [-p {fp32,fp16,int8}]
       [-t TEST_CASES]
       [-r TEST_RUNS]
       [--verbose]
       [-e]
       [--overwrite]
       [--use_int64_inputs]
       [-s {0,1,2}]
       [-a]
       [--keep_io_types]
       [--io_block_list IO_BLOCK_LIST [IO_BLOCK_LIST ...]]
       [--op_block_list OP_BLOCK_LIST [OP_BLOCK_LIST ...]]
       [--node_block_list NODE_BLOCK_LIST [NODE_BLOCK_LIST ...]]
       [--force_fp16_initializers]
convert_to_onnx.py: error: argument -t/--test_cases: expected one argument
```

> **NOTE**: Since the versions given in the notebook were not working (as in issue 1️⃣), I installed all of those versions in the newer state. And hence, I was able to generate the second issue. Thus the second issue was generated after using the versions like given below:
```sh
!{sys.executable} -m pip install install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cpu
!{sys.executable} -m pip install flatbuffers >>pip_output.txt

!{sys.executable} -m pip install onnxruntime
!{sys.executable} -m pip install transformers onnx psutil pytz py-cpuinfo py3nvml netron
```

___
> Running the code on Colab. Please update the code/information.

Thank you 🙏🏻

### Page / URL

https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/transformers/notebooks/Inference_GPT2_with_OnnxRuntime_on_CPU.ipynb

**Comment by tianleiwu** [comment]
@AayushSameerShah. 
Thanks for the feedback. The notebook is out of date. We'll update it and let you know once it is done.

**Comment by AayushSameerShah** [comment]
Hie, @tianleiwu 👋🏻 
Thanks for the response. I would also like to know a couple of things if you don't mind:

### 1️⃣ Does `onnxruntime` provide a way to **generate text** from the given text automatically?
The reason behind asking is the library *only performs the forward pass* and gives the `logits` to the program. Now it is on the developer to "make sense" of these logits by converting them back to the actual token.

> ❔ So, Does the library provide any kind of automatic conversion?

### 2️⃣ How to get the tokenizer?
In python we use the huggingface tokenizer to convert raw text into the tokens, but **in java** I am not sure what to use. And also since *my assumption* from the first question above is to manually decode, we also need a tokenizer to decode the numbers into tokens.
*(Yeah, I've got to tell you that I am using this library to run inference in Java)*

> ❔  So, how to handle tokenizer in java? Does this library provide such thing?

### 3️⃣ Can we set generation hyperparameters like temperature, topK etc?
Again from the first question's assumption, I am not sure **whether this library provides** any way to control the generation. Or do we need to manually create the sampling and temperature logic?

> ❔ So, Does this library provide the generation hyperparameters?

___
Please direct me,
Thank you so much 🙏🏻

**Comment by tianleiwu** [comment]
@AayushSameerShah, 

(1) There are some other operators like BeamSearch, GreedySearch and Sampling etc for text generation, those operator will output word IDs so it is easy to converting them back to token given a dictionary. 
See https://github.com/microsoft/onnxruntime/blob/13cc6192e5d877d7c782eaaa7f9fd1544a1902f2/onnxruntime/core/graph/contrib_ops/contrib_defs.cc#L1065-L1189
and https://github.com/microsoft/onnxruntime/blob/13cc6192e5d877d7c782eaaa7f9fd1544a1902f2/onnxruntime/python/tools/transformers/convert_generation.py for example.

(2) There are some tokenizer support in onnxruntime extension. See https://github.com/microsoft/onnxruntime-extensions/tree/main/operators/tokenizer. I think you can use it in Jave since the code is compiled into DLL.

(3) BeamSearch support some parameters like temperature, but top-k right now.
